source/extensions/filters/network/thrift_proxy/twitter_protocol_impl.cc

```diff
@@ -1031,6 +1036,8 @@ void TwitterProtocolImpl::updateMetadataWithRequestHeader(const ThriftObject& he
   // objects
   for (const auto& delegation : *req_header.delegations()) {
     std::string key = fmt::format(":d:{}", delegation.src_);
+    // LowerCaseString doesn't allow '\0', '\n', and '\r'.
+    key = absl::StrReplaceAll(key, {{std::string(1, '\0'), ""}, {"\n", ""}, {"\r", ""}});
```

Comment:
nit: `const std::string key` and then just call `absl::StrReplaceAll()` on `delegation.src_`